packages/api/src/store/asset-table.ts

```diff
@@ -72,4 +73,21 @@ export default class AssetTable extends Table<WithID<Asset>> {
     }
     return assets[0];
   }
+
+  async findDuplicateUrlUpload(
+    url: string,
+    userId: string
+  ): Promise<WithID<Asset>> {
+    const createdAfter = Date.now() - DUPLICATE_ASSETS_THRESHOLD;
```

Comment:
Good idea, I think DUPLICATE_ASSETS_THRESHOLD saves us from most weird states (something ends up permanently in `waiting` / `processing` etc.)